staff/meetings/meetings.py

```diff
@@ -0,0 +1,175 @@
+    if state:
+        states = {k: v for k, v in states.items() if v == state}
+    return(states)
+
+
+def quorum():
+    bod = ls(state="bod")
+    return(int(ceil(2 / 3 * len(bod))))
+
+
+def get_template(choice):
+    minutes_folder = get_minutes_folder()
+    if exists(join(minutes_folder, choice, "template")):
+        return join(minutes_folder, choice, "template")
+    return join(minutes_folder, "template")
+
+
+def get_minutes_choices():
+    return(listdir(get_minutes_folder()))
+
+
+def get_current_minutes_path(choice):
+    return get_minutes_path(choice, get_semester())
+
+
+def get_minutes_file():
+    return strftime("%Y-%m-%d")
+
+
+def get_minutes_path(choice, semester):
+    path = join(get_minutes_folder(), choice)
+    if not exists(path):
+        raise ValueError("argument must be from get_minutes_choices()")
+    path = join(path, semester)
+    if not exists(path):
+        makedirs(path)
+    return path
+
+
+def get_semester():
+    now = datetime.now()
+    year = now.year
+    month = now.month
+
+    AUG = 9
+    MAY = 5
+    # a reasonable approximation for semesters
+    if month > AUG or (month == AUG and day > 20):
+        sem = "Fall"
+    elif month < MAY or (month == MAY and day < 20):
+        sem = "Spring"
+    else:
+        sem = "Summer"
+    return(join(str(year), sem))
+
+
+def get_minutes(folder):
+    # checks that file is ####.##.## exactly
+    # this is done to excluse extra files that uploaded like pdfs and
+    # membership file
+    pattern = compile("^[0-9]{4}-[0-9]{2}-[0-9]{2}$")
+    return(sorted([i for i in listdir(folder) if pattern.match(i)]))
+
+
+def attendance(path):
+    attended = []
+    with open(path) as f:
+        lines = f.read().splitlines()
+        i = lines.index("Attendance:") + 1
+        while lines[i] and len(lines[i].split()) == 1:
+            attended += [lines[i]]
+            i += 1
+    return(attended)
+
+
+def new_semester():
+    year, semester = get_semester().split("/")
+    sem_map = {"Spring": "Fall", "Fall": "Spring"}
+    year_map = {"Spring": -1, "Fall": 0}
+    last_sem = join(str(int(year) + year_map[semester]),
+                    sem_map[semester])
+    old_file = join(get_minutes_path("bod", last_sem), "membership")
+    new_file = join(get_current_minutes_path("bod"), "membership")
+    copyfile(old_file, new_file)
+
+
+def get_membership_status(folder):
+    path = join(folder, "membership")
+    if not exists(path):
+        new_semester()
+    status = defaultdict(lambda: 0)
+    with open(path) as f:
+        for line in f:
+            user, state = line.split()
+            status[user] = state
+    return status
+
+
+def minutes_setup(notes, choice):
+    if not exists(notes):
+        copyfile(get_template(choice), notes)
+    with open(notes, "r") as f:
+        s = f.read()
+    s = s.replace("<USERNAME>", getuser())
+    t = strftime("%H:%M")
+    s = s.replace("<START TIME>", t)
```

Comment:
i probably would have had the template be a large format string, e.g.:

```
Meeting

Attendance:

Meeting Started at {start_time}

Meeting Adjourned at {end_time}

Notetaker: {notetaker}
```

then you can do

```python
s = s.format(
    notetaker=getuser.getuser(),
    start_time=...,
    end_time=...,
)
```

which is a bit nicer than implementing your own thing with str.replace